### PR TITLE
Use python logging framework for logging

### LINKED ISSUE #171: Print debugging information on STDERR
Conventionally, programs print output not related to their main product on STDERR. Calling `mbedls --debug` prints debugging output on STDOUT.

In some situations (debugging) we want to see verbose output from mbedls that explains what is going on, yet use the list of the detected boards for something useful. STDIO separation is ought to provide for this feature.

### PR DESCRIPTION
**Pull request by theotherjimmy**
The title really says it all

Resolves #171

### CODE REVIEW

**Comment by theotherjimmy** [comment]
@bridadan Could you take a look?

File: mbed_lstools/lstools_base.py [hunk 1]
```diff
@@ -47,7 +49,7 @@ def wrapper(self, *args):
                 raise e
             lock.release()
         else:
-            self.err("timed_mbedls_lock", "Failed to acquired mock file lock in %d seconds!" % timeout)
+            logger.err("Failed to acquired mock file lock in %d seconds!" , timeout)
```
Comment (bridadan) [inline mbed_lstools/lstools_base.py]: @theotherjimmy I don't _think_ the `logger.err` is an actual function in `logging`. I think it's actually `logger.error`.

I'm going off this doc page: https://docs.python.org/2/library/logging.html#logging.Logger.error

Comment (theotherjimmy) [reply mbed_lstools/lstools_base.py]: You're correct.

**Comment by theotherjimmy** [comment]
Fixed. Could you review again @bridadan

File: mbed_lstools/lstools_base.py [hunk 2]
```diff
@@ -47,7 +49,7 @@ def wrapper(self, *args):
                 raise e
             lock.release()
         else:
-            self.err("timed_mbedls_lock", "Failed to acquired mock file lock in %d seconds!" % timeout)
+            logger.error("Failed to acquired mock file lock in %d seconds!" , timeout)
```
Comment (bridadan) [inline mbed_lstools/lstools_base.py]: Uber nit: there's an extra space between the comma and the first parameter.

File: mbed_lstools/lstools_base.py [hunk 3]
```diff
@@ -790,9 +766,9 @@ def mount_point_ready(self, path):
         result = os.path.exists(path)
 
         if result:
-            self.debug(self.mount_point_ready.__name__, "Mount point %s is ready" % path)
+            logger.debug("Mount point %s is ready", path)
         else:
-            self.debug(self.mount_point_ready.__name__, "Mount point %s does not exist" % (path))
+            logger.debug("Mount point %s does not exist" , path)
```
Comment (bridadan) [inline mbed_lstools/lstools_base.py]: Uber nit: there's an extra space between the first parameter and the comma.

**Comment by theotherjimmy** [comment]
@bridadan Another round!

**Comment by theotherjimmy** [comment]
Uber nits are fine by me. I would just end up fixing them later.

**Pull request merged**